server/src/main/java/org/opensearch/search/startree/StarTreeTraversalUtil.java

```diff
@@ -40,16 +42,20 @@
  *  @opensearch.experimental
  *  @opensearch.internal
  */
-public class StarTreeFilter {
-    private static final Logger logger = LogManager.getLogger(StarTreeFilter.class);
+public class StarTreeTraversalUtil {
```

Comment:
nit : subjective but Since this is the core algorithm - I don't want this to be called util - can we rename it to something else ? StarTreeTraverser/StarTreeIndexQuerier or something better

Reply:
Forgot adding a TODO 
This will become a interface later and one of the non-default implementations might be Range Aggregations. 
Named it Util to indicate it's stateless. 
Will wait for others to chime in as well.